[이진우] sprint11

ajantang opened the pull request:
## 요구사항

### 기본

- [x] Github에 위클리 미션 PR을 만들어 주세요.
- [x] React 및 Express를 사용해 진행합니다.
- [x] TypeScript를 활용해 프로젝트의 필요한 곳에 타입을 명시해 주세요.
- [x] any 타입의 사용은 최소화해 주세요.
- [x] 복잡한 객체 구조나 배열 구조를 가진 변수에 인터페이스 또는 타입 별칭을 사용하세요.
- [x] Union, Intersection, Generics 등 고급 타입을 적극적으로 사용해 주세요.
- [x] 타입 별칭 또는 유틸리티 타입을 사용해 타입 복잡성을 줄여주세요.
- [x] 타입스크립트 컴파일러가 에러 없이 정상적으로 작동해야 합니다.

### 프론트엔드

- [x] 기존 React(혹은 Next) 프로젝트를 타입스크립트 프로젝트로 마이그레이션 해주세요.
- [x] TypeScript를 활용해 프로젝트의 필요한 곳에 타입을 명시해 주세요.

## 주요 변경사항

- 초기 업로드
-

## 스크린샷

![image](이미지url)

## 멘토에게

- 현재 TS 적용 중입니다
- 기존의 코드에 (미완성 || 만족 못하는) 부분들이 많아서 수정 중입니다

ajantang committed:
feat : change api of post & post-comment & auth

BE를 sprint10 미션용으로 수정

참고 : user, product 등 다른 api도 수정 예정

ajantang committed:
chore : install typescript & type package

ajantang committed:
chore : add config setting for ts

ajantang committed:
feat : exchange article to post

article -> post, title -> name

참고 : ts 변환 과정이라 수정 부분 테스트 추후 진행 예정

ajantang committed:
feat : change comment list api function

BE 변경 사항에 맞춰 수정

ajantang committed:
feat : exchange to ts code (unfinished)

ajantang committed:
feat : exchange js code to ts code (temp)

일단 실행 후 코드 / 로직 수정해야할 부분이 보여서 임시로 작성한 ts 코드 사용

ajantang committed:
feat : add home page components

홈페이지를 구성하는 component 추가

참고 : 반응형 적용 안함. tailwind css 반응형 설정. 기존의 방법 이외에 선호되는 방법을 찾아서 적용 예정

ajantang committed:
style : delete css files

일부 새로 생성한 css 파일 삭제
반응형 css 적용을 위해 3개지 선택 사항 중 1안 선택
1. tsx에 직접 tailwind css 사용
2. css module에 tailwind + vanilla(@media) 조합 사용
3. tailwind.config.js만 사용

ajantang committed:
style : delete css files

일부 새로 생성한 css 파일 삭제

wildCodingWarrior commented on src/app/bulletin-board/PostPreview.tsx:
```diff
@@ -0,0 +1,69 @@
+  favoriteCount = 0,
+  createdDate,
+}: PostPreviewProps) {
+  const postPreviewClass = `${style["post-preview"]}`;
+  const postPreviewTopBarClass = `flex flex-row justify-between ${style["top-bar"]}`;
+  const postPreviewNameClass = `font-semibold ${style.name}`;
+  const postPreviewBottomBarClass = `flex flex-row justify-between ${style["bottom-bar"]}`;
+  const postPreviewBottomBarNicknameDateClass = `flex flex-row items-center ${style["bottom-bar-nickname-date"]}`;
+  const postPreviewBottomNicknameClass = `font-normal ${style["nickname-name"]}`;
+
```
이런식으로 정리하시면 tailwind CSS intellisense 혜택을 못보지 않나요?

차라리 조금 공부하더라도 tailwind-merge랑 clsx를 활용하는 cn유틸을 사용해보는 것이 어떤가 싶습니다.

ajantang replied:
아직 tailwind를 잘 못 쓰고 있습니다. 정확히는 tailwind 적용한 코드를 돌아보니 tailwind.config.js에 몰아서 정의하다보니, 오히려 관리의 어려움을 느껴, 현재는 tsx 파일에 직접 tailwind를 사용하는 방식으로 수정 중입니다.
tailwind-merge랑 clsx를 활용하는 cn유틸은 자료를 찾아보겠습니다.

참고 :  아마 위의 코드는 바닐라로 만들고 아직 tailwind로 완전히 적용 안 한 부분일 겁니다.

wildCodingWarrior replied:
tailwind를 사용하는 방법에 대해서는 내일 자세히 한 번 알려드리겠습니다.

wildCodingWarrior commented on src/types/components.ts:
```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from "react";
+
+export interface MiddleBannerProps {
+  isLeftImage: boolean;
+  imagePath: string;
+  textSetWidth: string;
+  topText: string;
+  middleText: string;
+  bottomText: string;
+}
+
+export interface SignBottomTextProps {
+  message: string;
+  linkText: string;
+  linkPath: string;
+}
+
+export interface DropdownProps {
+  dropdwonClass?: string;
+  minimise?: string | boolean;
+  children: ReactNode;
```
컴포넌트에 해당하는 Props는 해당 컴포넌트에 작성하는게 더 응집도 있는 코드라고 생각합니다.

ajantang replied:
알겠습니다. 수정해두겠습니다.

wildCodingWarrior commented on package.json:
```diff
@@ -3,32 +3,46 @@
-    "dev": "cross-env NODE_OPTIONS='--inspect' next dev",
-    "build": "next build",
+    "dev": "cross-env NODE_OPTIONS=development next dev",
+    "build": "next build && tsc",
```
이 tsc는 없어도 무방하겠습니다

ajantang replied:
수정해두겠습니다

wildCodingWarrior commented on package.json:
```diff
@@ -3,32 +3,46 @@
     "react-dom": "^18",
     "react-hook-form": "^7.53.0",
     "react-modal": "^3.16.1",
+    "react-query": "^3.39.3",
```
이 패키지 옛날 버전이라서 @tanstack/react-query를 사용해보세요.

ajantang replied:
"@tanstack/react-query": "^5.56.2", 를 사용 중입니다

wildCodingWarrior reviewed (approved):
고생하셨습니다!

ajantang committed:
style : divide input components for sign up

회원 가입에 사용되는 컴포넌트 분리
ts에 적합한 코드로 정리

ajantang committed:
refactor : locate props type in individual component files

ajantang committed:
refactor : delete useless script

ajantang committed:
style : arrange header & footer code

ajantang committed:
feat : add temp code for pagination

SSR에서 호출하는 API가 로컬스토리지에 토큰에 접근하는 것을 방지

참고 : 토큰을 로컬스토리지에 저장한 것은 임시 코드여서 이 부분을 어떻게 수정할지 고민 필요

ajantang committed:
feat : save token by using cookie

ajantang committed:
feat : add temp logic for cookie

임시로 상품 리스트 / 상품 상세 조회 페이지에 적용할 API 코드 작성

참고 : 테스트 과정에서 일단 문제가 없어보여 커밋. 다른 API 함수 전부 수정해야 됨

ajantang committed:
feat : add best product list

ajantang committed:
feat : add product comment maker

ajantang committed:
feat : change path of product comment list

ajantang committed:
feat : change orderby of product comment list & delete log

ajantang committed:
feat : add post list

ajantang committed:
style : delete css module of post list

wildCodingWarrior commented on src/app/bulletin-board/PostList.tsx:
```diff
@@ -24,29 +24,76 @@ import {
-  const boardTopBarBtnWritePostClass = `${style["top-bar-btn-write-post"]}`;
-
-  const boardMiddleBarClass = `flex flex-row justify-between ${style["middle-bar"]}`;
+  const boardClass = classNames(
+    "w-full",
+    "mt-[4rem]",
+    "mb-[29.3rem]",
+    "ta:mt-[2.4rem]",
+    "ta:mb-[1.9rem]",
+    "mo:mt-[2.4rem]",
+    "mo:mb-[9.1rem]"
+  );
+  const boardTopBarClass = classNames(
+    "flex",
+    "flex-row",
+    "items-center",
+    "justify-between",
+    "w-full",
+    "h-[4.2rem]"
+  );
```
classNames 쓴것도 좋네요.

wildCodingWarrior reviewed (approved):
고생하셨습니다!

ajantang committed:
style : replace best post module css to tailwind

ajantang committed:
fix : import error(temp)

배포 에러 해결을 위한 임시 코드 수정

참고 : 커스텀으로 만든 Input, TextArea와 그걸 기반으로 하는 컴포넌트 수정 예정

ajantang committed:
fix : delete unused props
